### PR TITLE
Add utility to escape a regex string

### LINKED ISSUE #66: Implement `escape-regexp-string`
See [utils-escape-regex-string](https://github.com/kgryte/utils-escape-regex-string). `@stdlib/utils/escape-regexp-string`.

### PR DESCRIPTION
**Pull request by Planeshifter**
Resolves #66 .

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

* [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
* [x] Read and understand the [Code of Conduct][code-of-conduct].
* [x] Read and understood the [licensing terms][license].
* [x] Searched for existing issues and pull requests __before__ submitting this pull request.
* [x] Filed an issue (or an issue already existed) __prior to__ submitting this pull request.
* [x] Rebased onto latest `develop`.
* [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:

* contains a straight-forward port of [utils-escape-regex-string](https://github.com/kgryte/utils-escape-regex-string)

## Related Issues

> Does this pull request have any related issues?

This pull request:

* resolves #
* fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

---

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/master/CONTRIBUTING.md
[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/master/CODE_OF_CONDUCT.md
[license]: https://github.com/stdlib-js/stdlib/blob/master/LICENSE

### CODE REVIEW

File: lib/node_modules/@stdlib/utils/escape-regex-string/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "scripts": {},
+  "main": "./lib",
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/stdlib-js/stdlib.git"
+  },
+  "homepage": "https://github.com/stdlib-js/stdlib",
+  "keywords": [
+    "stdlib",
+    "{{TODO:keywords}}"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/escape-regex-string/package.json]: Forgot keywords.

File: lib/node_modules/@stdlib/utils/escape-regex-string/lib/escape_regex_string.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+* Escapes a regular expression string.
+*
+* @param {string} str - regular expression string
+* @throws {TypeError} first argument must be a string primitive
+* @returns {string} escaped string
+*
+* @example
+* str = rescape( '[A-Z]*' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/escape-regex-string/lib/escape_regex_string.js]: Missing `var`.

File: lib/node_modules/@stdlib/utils/escape-regex-string/lib/escape_regex_string.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		len = str.length;
+		for ( i = len-1; i >= 0; i-- ) {
+			if ( str[ i ] === '/' ) {
+				break;
+			}
+		}
+	}
+	// If we searched the string to no avail or if the first letter is not `/`, assume that the string is not of the form `/[...]/[gimy]`:
+	if ( i === void 0 || i <= 0 ) {
+		return str.replace( RE, '\\$&' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/escape-regex-string/lib/escape_regex_string.js]: Use `@stdlib/string/replace`.

File: lib/node_modules/@stdlib/utils/escape-regex-string/lib/escape_regex_string.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	}
+	// We need to de-construct the string...
+	s = str.substring( 1, i );
+
+	// Only escape the characters between the `/`:
+	s = s.replace( RE, '\\$&' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/escape-regex-string/lib/escape_regex_string.js]: Use `@stdlib/string/replace`.

**Review by kgryte — CHANGES_REQUESTED** [review]

**Comment by Planeshifter** [comment]
Thank you! Addressed the problems in the most recent commits.

**Comment by Planeshifter** [comment]
One more question: Are we fine with the name `rescape`? 
If so, I will add the module to the namespace and then we can merge.

**Comment by kgryte** [comment]
Yes, `rescape` is fine.

**Review by kgryte — APPROVED** [review]

**Comment by kgryte** [comment]
Once Circle CI gives the green light, feel free to merge.

**Pull request merged**